### PR TITLE
1703 4 hard delete attachments

### PR DESCRIPTION
**Pull request by BCerki**
card: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1703

The card is only about hard deleting attachments in the first revision. I think there may be some other cases in which we'd like to hard delete (e.g., deleting an attachment that was uploaded in the same form change) and made a tech debt ticket to investigate: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1776

Also, hard deleting attachments as part of discarding a project revision was split into another card: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1798

Also also, cypress CI upload and deletion will be another card: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1839

### CODE REVIEW

**Comment by BCerki** [comment]
@dleard this is the attachments PR, and this is where Gurjeet fixed the middleware: https://github.com/bcgov/cas-cif/pull/1838

**Review by Sepehr-Sobhani — CHANGES_REQUESTED** [review]
Awesome! I just have a few leftover things and some curiosity questions :) Once the attachment upload feature is fixed, I would love to manually test this function.

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,5 @@ cypress.env.json
 app/cypress.env.json
 dev/db/data
 **/.DS_Store
+
+gha-creds-*.json
```
[the remark this answers is not in the document]
Comment (BCerki) [reply .gitignore:17]: This is leftover from experimenting with CI, will remove

File: schema/deploy/mutations/discard_project_attachment_form_change.sql [hunk 2]
```diff
@@ -2,27 +2,23 @@
 
 begin;
 
-create or replace function cif.discard_project_attachment_form_change(form_change_id int)
-returns setof cif.form_change
+drop function cif.discard_project_attachment_form_change;
+create function cif.discard_project_attachment_form_change(form_change_id int)
+returns cif.form_change
```
[the remark this answers is not in the document]
Comment (BCerki) [reply schema/deploy/mutations/discard_project_attachment_form_change.sql:7]: If the return type is different you can't replace, you have to drop and remake

[the remark this answers is not in the document]
Comment (Sepehr-Sobhani) [reply schema/deploy/mutations/discard_project_attachment_form_change.sql:7]: Woooo, good to know 👍

**Comment by BCerki** [comment]
work finished in #1843

**Pull request closed without merging**